db/db_utils.go

```diff
@@ -170,8 +171,30 @@ func UpdateId(collection string, id, update interface{}) error {
 }
 
 // UpdateAll updates all matching documents in the collection.
-func UpdateAll(collection string, query interface{},
-	update interface{}) (*mgo.ChangeInfo, error) {
+func UpdateAll(collection string, query interface{}, update interface{}) (*mgo.ChangeInfo, error) {
+	switch q := query.(type) {
+	case *Q, Q:
+		grip.EmergencyPanic(message.Fields{
+			"message":    "invalid query passed to update all",
+			"cause":      "programmer error",
+			"query":      query,
+			"collection": collection,
+		})
+	case nil:
+		grip.EmergencyPanic(message.Fields{
+			"message":    "nil query passed to update all",
+			"query":      query,
+			"collection": collection,
+		})
+	case map[string]interface{}:
+		if len(q) == 0 {
```

Comment:
Is it not reasonable that a user may want to update every document in a collection?